main.go

```diff
@@ -1,80 +1,82 @@
 package main
 
 import (
-	health "github.com/Financial-Times/go-fthealth/v1_1"
-	status "github.com/Financial-Times/service-status-go/httphandlers"
+	fw "github.com/Financial-Times/content-collection-unfolder/forwarder"
+	prod "github.com/Financial-Times/content-collection-unfolder/producer"
+	res "github.com/Financial-Times/content-collection-unfolder/resolver"
+	"github.com/Financial-Times/message-queue-go-producer/producer"
 	log "github.com/Sirupsen/logrus"
 	"github.com/jawher/mow.cli"
+	"net"
 	"net/http"
 	"os"
-	"os/signal"
-	"syscall"
+	"time"
 )
 
 const appDescription = "UPP Service that forwards mapped content collections to the content-collection-rw-neo4j. If a 200 answer is received from the writer, it retrieves the elements in the collection from the document-store-api and places them in Kafka on the Post Publication topic so that notifications will be created for them."
 
 func main() {
 	app := cli.App("content-collection-unfolder", appDescription)
-
-	appSystemCode := app.String(cli.StringOpt{
-		Name:   "app-system-code",
-		Value:  "content-collection-unfolder",
-		Desc:   "System Code of the application",
-		EnvVar: "APP_SYSTEM_CODE",
-	})
-
-	appName := app.String(cli.StringOpt{
-		Name:   "app-name",
-		Value:  "Content Collection Unfolder",
-		Desc:   "Application name",
-		EnvVar: "APP_NAME",
-	})
-
-	port := app.String(cli.StringOpt{
-		Name:   "port",
-		Value:  "8080",
-		Desc:   "Port to listen on",
-		EnvVar: "APP_PORT",
-	})
+	sc := createServiceConfiguration(app)
 
 	log.SetLevel(log.InfoLevel)
-	log.Infof("[Startup] content-collection-unfolder is starting ")
 
 	app.Action = func() {
-		log.Infof("System code: %s, App Name: %s, Port: %s", *appSystemCode, *appName, *port)
-
-		go func() {
-			serveAdminEndpoints(*appSystemCode, *appName, *port)
-		}()
-
-		// todo: insert app code here
-
-		waitForSignal()
+		log.Infof("[Startup] content-collection-unfolder is starting with service config %v", sc.toMap())
+
+		client := setupHttpClient()
+		producer := setupMessageProducer(sc, client)
+
+		hc := &healthConfig{
+			appDesc:                  appDescription,
+			port:                     *sc.appPort,
+			appSystemCode:            *sc.appSystemCode,
+			appName:                  *sc.appName,
+			writerHealthURI:          *sc.writerHealthURI,
+			contentResolverHealthURI: *sc.contentResolverHealthURI,
+			producer:                 producer,
+			client:                   client,
+		}
+
+		newRouting(
```

Comment:
I found the naming here a bit confusing. Maybe using newRouter() would help it a bit, or even splitting this function call in two steps..